src/api/controllers/db-controller.ts

```diff
@@ -778,17 +823,79 @@ export async function getMempoolTxFromDataStore(
   if (!mempoolTxQuery.found) {
     return { found: false };
   }
-  const parsedMempoolTx = parseDbMempoolTx(mempoolTxQuery.result);
+  let parsedMempoolTx = parseDbMempoolTx(mempoolTxQuery.result);
   // If tx type is contract-call then fetch additional contract ABI details for a richer response
   if (parsedMempoolTx.tx_type === 'contract_call') {
-    await getContractCallMetadata(db, mempoolTxQuery.result, parsedMempoolTx);
+    const transaction = await getContractCallMetadata(
+      db,
+      parseDbMempoolTx(mempoolTxQuery.result) as MempoolContractCallTransaction,
+      mempoolTxQuery.result
+    );
+    if (transaction) {
+      parsedMempoolTx = transaction as MempoolContractCallTransaction;
+    }
```

Comment:
Instead of repeating logic here, why don't you call the new `getMempoolTxsFromDataStore` and validate/return the first result?

Reply:
Small clarification required - are you recommending that I use the same function for an individual transaction and for a list of transactions?

Reply:
Yes. Here, in the individual tx function, you'd call `getMempoolTxsFromDataStore` but with an array of only one tx (the one given as arg) and then take the first result to return.